auctioneer/client.go

```diff
@@ -431,6 +431,20 @@ func (c *Client) SubmitOrder(ctx context.Context, o order.Order,
 		MaxBatchFeeRateSatPerKw: uint64(o.Details().MaxBatchFeeRate),
 	}
 
+	details.AllowedNodeIds = make(
+		[][]byte, len(o.Details().AllowedNodeIDs),
+	)
+	for idx, nodeID := range o.Details().AllowedNodeIDs {
+		details.AllowedNodeIds[idx] = nodeID[:]
```

Comment:
Not sure if this is safe? Since `nodeID` is restricted to the scope of the for loop.

Reply:
Good intuition, In Go, the runtime is the one that decides what goes in the stack/heap (the specification does not define it). So, even for pointers, the obj survives the scope.

Reply:
I actually ran a little experiment, because I wasn't quite sure myself. PTAL: https://go.dev/play/p/geeXf0C5E6m

Reply:
ohhh,  you are right. The var will out live the loop but all the elements in the slice of `[]byte` will be equal to the last one!
:pray: Thank you for catching this @bhandras

Reply:
It's tricky and I think the compiler/linter should be more vocal about this as it's easy to miss.

Reply:
Oh wow, good catch! I wouldn't have caught that one :open_mouth: